pkg/cephfs/fsjournal.go

```diff
@@ -46,52 +46,44 @@ request name lock, and hence any stale omaps are leftovers from incomplete trans
 hence safe to garbage collect.
 */
 func checkVolExists(ctx context.Context, volOptions *volumeOptions, secret map[string]string) (*volumeIdentifier, error) {
-	var (
-		vi  util.CSIIdentifier
-		vid volumeIdentifier
-	)
+	var vid volumeIdentifier
 
 	cr, err := util.NewAdminCredentials(secret)
 	if err != nil {
 		return nil, err
 	}
 	defer cr.DeleteCredentials()
 
-	imageUUID, err := volJournal.CheckReservation(ctx, volOptions.Monitors, cr,
+	imageData, err := volJournal.CheckReservation(ctx, volOptions.Monitors, cr,
 		volOptions.MetadataPool, volOptions.RequestName, volOptions.NamePrefix, "", "")
 	if err != nil {
 		return nil, err
 	}
-	if imageUUID == "" {
+	if imageData == nil {
 		return nil, nil
 	}
-
-	// now that we now that the reservation exists, let's get the volume name from
-	// the omap
-	_, vid.FsSubvolName, _, _, err = volJournal.GetObjectUUIDData(ctx, volOptions.Monitors, cr,
-		volOptions.MetadataPool, imageUUID, false)
-	if err != nil {
-		return nil, err
-	}
+	imageUUID := imageData.ImageUUID
+	vid.FsSubvolName = imageData.ImageAttributes.ImageName
 
 	_, err = getVolumeRootPathCeph(ctx, volOptions, cr, volumeID(vid.FsSubvolName))
 	if err != nil {
 		if _, ok := err.(ErrVolumeNotFound); ok {
-			err = volJournal.UndoReservation(ctx, volOptions.Monitors, cr, volOptions.MetadataPool, vid.FsSubvolName, volOptions.RequestName)
+			err = volJournal.UndoReservation(ctx, volOptions.Monitors, cr, volOptions.MetadataPool,
+				volOptions.MetadataPool, vid.FsSubvolName, volOptions.RequestName)
 			return nil, err
 		}
 		return nil, err
 	}
+
+	// check if topology constraints match what is found
+	// TODO: we need an API to fetch subvolume attributes (size/datapool and others), based
+	// on which we can evaluate which topology this belongs to.
+	// TODO: CephFS topology support is postponed till we get the same
```

Comment:
is this true? because I see template  changes for cephfs also

Reply:
Everything else is in place, **except** the ability to inspect an subvolumes data pool. Hence, leaving it in place and will cover support/feature situation for CephFS in documentation.